### PR TITLE
Implement never force unwrap

### PR DESCRIPTION
**Pull request by LaurenWhite**

### CODE REVIEW

File: tools/swift-format/Sources/Rules/NeverForceUnwrap.swift [hunk 1]
```diff
@@ -8,5 +8,35 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#force-unwrapping-and-force-casts
 public final class NeverForceUnwrap: SyntaxLintRule {
+  
+  // Checks if "XCTest" is an import statement
+  public override func visit(_ node: ImportDeclSyntax) {
+    var iterator = node.path.makeIterator()
+    while let component = iterator.next() {
+      if component.name.text == "XCTest" {
+        context.importsXCTest = true
```
Comment (harlanhaskins) [inline tools/swift-format/Sources/Rules/NeverForceUnwrap.swift]: Ditto from NeverUseImplicitlyUnwrappedOptionals. Would like to see this cached on Context and see the visitor deduplicated.

File: tools/swift-format/Sources/Rules/NeverForceUnwrap.swift [hunk 2]
```diff
@@ -8,5 +8,35 @@ import SwiftSyntax
...
+  public override func visit(_ node: AsExprSyntax) {
+    guard !context.importsXCTest else { return }
+    print(node.typeName.description)
+    diagnose(.doNotForceCast(name: node.typeName.description), on: node)
+  }
+}
 
+extension Diagnostic.Message {
+  static func doNotForceUnwrap(name: String) -> Diagnostic.Message {
+    return .init(.warning, "Do not force unwrap '\(name)'")
```
Comment (harlanhaskins) [inline tools/swift-format/Sources/Rules/NeverForceUnwrap.swift]: Uncapitalize diagnostic messages, please.

**Comment by harlanhaskins** [comment]
Small changes but looks good!

File: tools/swift-format/Tests/SwiftFormatTests/DontRepeatTypeInStaticPropertiesTests.swift [hunk 3]
```diff
@@ -0,0 +1,8 @@
+//
```
Comment (allevato) [inline tools/swift-format/Tests/SwiftFormatTests/DontRepeatTypeInStaticPropertiesTests.swift]: Can you remove this file from this PR?

File: tools/swift-format/Tests/SwiftFormatTests/GroupNumericLiteralsTests.swift [hunk 4]
```diff
@@ -0,0 +1,8 @@
+//
```
Comment (allevato) [inline tools/swift-format/Tests/SwiftFormatTests/GroupNumericLiteralsTests.swift]: Also this one.

Otherwise, it's ready to merge.

**Pull request merged**